### PR TITLE
Upload img

### LINKED ISSUE #7: Upload image
Upload image process

Estimation : 3h

### PR DESCRIPTION
**Pull request by LFZDavid**

### CODE REVIEW

File: src/Entity/Media.php [hunk 1]
```diff
@@ -22,6 +23,8 @@ class Media
      */
     private $url;
```
Comment (crazymad) [inline src/Entity/Media.php:23]: ?string

File: src/Entity/Media.php [hunk 2]
```diff
@@ -22,6 +23,8 @@ class Media
...
 
+    private $file;
```
Comment (crazymad) [inline src/Entity/Media.php:26]: ?UploadedFile

File: src/Service/FileUploader.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
+class FileUploader
+{
+    private $targetDirectory;
+    private $slugger;
+
+    public function __construct($targetDirectory, SluggerInterface $slugger)
```
Comment (crazymad) [inline src/Service/FileUploader.php:14]: targetDirectory is a string

File: src/Service/FileUploader.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+namespace App\Service;
+
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
+class FileUploader
+{
+    private $targetDirectory;
+    private $slugger;
```
Comment (crazymad) [inline src/Service/FileUploader.php:12]: SluggerInterface

File: src/Service/FileUploader.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace App\Service;
+
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
+class FileUploader
+{
+    private $targetDirectory;
```
Comment (crazymad) [inline src/Service/FileUploader.php:11]: string

File: src/Service/FileUploader.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+        try {
+            $file->move($this->getTargetDirectory(), $fileName);
+        } catch (FileException $e) {
+
+            return $e->getMessage();
+            //todo : handle exception and display message in front
+        }
+
+        return '/img/tricks/'.$fileName;
```
Comment (crazymad) [inline src/Service/FileUploader.php:34]: meh. I understand why, but then if you change the parameter in the config, it won't work here anymore

File: src/Service/FileUploader.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function upload(UploadedFile $file)
+    {
+        $originalFilename = pathinfo($file->getClientOriginalName(), PATHINFO_FILENAME);
+        $safeFilename = $this->slugger->slug($originalFilename);
+        $fileName = $safeFilename.'-'.uniqid().'.'.$file->guessExtension();
+
+        try {
+            $file->move($this->getTargetDirectory(), $fileName);
```
Comment (crazymad) [inline src/Service/FileUploader.php:27]: $this->targetDirectory
why a getter ?

**Review by crazymad — CHANGES_REQUESTED** [review]
Good job on this !!

File: src/Entity/Media.php (hunk 1 above)
Comment (LFZDavid) [reply src/Entity/Media.php:23]: fixed on media-video branch

File: src/Entity/Media.php (hunk 2 above)
Comment (LFZDavid) [reply src/Entity/Media.php:26]: fixed on media-video branch

File: src/Service/FileUploader.php (hunk 3 above)
Comment (LFZDavid) [reply src/Service/FileUploader.php:14]: fixed on media-video branch

File: src/Service/FileUploader.php (hunk 4 above)
Comment (LFZDavid) [reply src/Service/FileUploader.php:12]: fixed on media-video branch

File: src/Service/FileUploader.php (hunk 5 above)
Comment (LFZDavid) [reply src/Service/FileUploader.php:11]: fixed on media-video branch

File: src/Service/FileUploader.php (hunk 7 above)
Comment (LFZDavid) [reply src/Service/FileUploader.php:27]: peasted from Symfony doc.
fixed on media-video branch

File: src/Service/FileUploader.php (hunk 6 above)
Comment (LFZDavid) [reply src/Service/FileUploader.php:34]: fixed on media-video branch

**Pull request merged**